std/array.d

```diff
@@ -1360,6 +1363,84 @@ pure nothrow bool sameTail(T)(in T[] lhs, in T[] rhs)
     }}
 }
 
+/**
+Checks whether a `slice` is part of a `whole` array.
+
+Params:
+
+    slice = The array to check for its origin
+    whole = The origin array to be queried
+
+Returns: `true` if `slice` is a part of `whole`.
+*/
+bool isSliceOf(T)(const scope T[] slice, const scope T[] whole) @trusted
```

Comment:
Use `in` instead of `const scope`. They are equivalent.

Reply:
Hehe, that used to be in the spec, but the compiler never did this:

- https://issues.dlang.org/show_bug.cgi?id=17928
- https://github.com/dlang/dlang.org/pull/2126
- https://github.com/dlang/phobos/pull/6074

Reply:
I've actually argued for years that `in` should never be used, because `scope` wasn't actually implemented for anything other than delegates, so using `in` was just risking having your code break as soon as `scope` was more fully implemented, but precisely because of that problem, Walter change `in` to just be `const`, meaning that `in` is now completely useless.

It's meaning may or may not change later, but either way, I think that it should be actively avoided unless/until it's actually implemented to mean something rather than just `const` and is clearly going to stay that way, at which point, the code can actually be designed with its proper meaning in mind instead of using `in` because of what it might mean in the future or because it seems like the opposite of `out`.

Reply:
Hmm, that's unfortunate. I've gotten used to using `in`.